packages/core/components/DataTable/helpers/getChartCellValue.ts

```diff
@@ -31,16 +31,19 @@ export const getChartCellValue = (row: string, column: string, config: TableConf
   let cellValue // placeholder for formatting below
   let labelValue = rowObj[column] // just raw X axis string
   if (column === config.xAxis?.dataKey) {
-    // not the prettiest, but helper functions work nicely here.
-    cellValue =
-      config.xAxis?.type === 'date'
-        ? formatDate(
-            config.table?.dateDisplayFormat || config.xAxis?.dateDisplayFormat,
-            parseDate(config.xAxis?.dateParseFormat, labelValue)
-          )
-        : labelValue
-    cellValue =
-      config.xAxis?.type === 'continuous' ? formatNumber(runtimeData[row][column], 'bottom', false, config) : labelValue
+    cellValue = labelValue
+    const isDateTime = config.xAxis?.type === 'date' || config.xAxis?.type === 'date-time'
+
+    if (isDateTime && config.table.dateDisplayFormat) {
+      cellValue = formatDate(config.table?.dateDisplayFormat, parseDate(config.table?.dateDisplayFormat, labelValue))
+    }
+
+    if (isDateTime && !config.table.dateDisplayFormat) {
+      cellValue = formatDate(config.xAxis?.dateDisplayFormat, parseDate(config.xAxis?.dateDisplayFormat, labelValue))
+    }

```

Comment:
make this more readable: 

`if (isDateTime) {
    const dateDisplayFormat = config.table?.dateDisplayFormat || 
    config.xAxis?.dateDisplayFormat;
    cellValue = formatDate(dateDisplayFormat, parseDate(dateDisplayFormat, labelValue)
}
`

Reply:
Fixed!

Reply:
This comment was for this and the if statement above it. You can handle both with these couple lines.

Reply:
@Atash3000

Reply:
Can you explain more please? I did not understood the issue here